manager/integration/tests/backupstore.py

```diff
@@ -116,14 +118,18 @@ def umount_nfs_backupstore(client, mount_path="/mnt/nfs"):
 
 
 def backupstore_cleanup(client):
-    backup_volumes = client.list_backup_volume()
-
-    # we delete the whole backup volume, which skips block gc
-    for backup_volume in backup_volumes:
-        delete_backup_volume(client, backup_volume.name)
-
-    backup_volumes = client.list_backup_volume()
-    assert backup_volumes.data == []
+    try:
+        backup_volumes = client.list_backup_volume()
+        # we delete the whole backup volume, which skips block gc
+        for backup_volume in backup_volumes:
+            delete_backup_volume(client, backup_volume.name)
+
+        backup_volumes = client.list_backup_volume()
+        assert backup_volumes.data == []
+    except longhorn.ApiError as e:
+        # when trying to list already deleted (nfs) backup volumes,
+        # it returns 500 on Server Error.
```

Comment:
Why can deleted backup volumes lead to NFS server error 500? You mean the backup deletion is still in progress and the lock blocks the response?

Reply:
It seems like an expected behavior when volume directory not exist on the nfs backstore?
```
longhorn.ApiError: (ApiError(...), '500 : error listing backups: error listing backup volumes: Failed to execute: /var/lib/longhorn/engine-binaries/c3y1huang-research-8-lh-engine-bcf73c2-dirty/longhorn [backup ls --volume-only nfs://longhorn-test-nfs-svc.default:/opt/backupstore], output Failed to execute: ls [-1 /var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes], output ls: cannot access \'/var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes\': No such file or directory\n, error exit status 2\n, stderr, time="2021-03-08T04:05:20Z" level=warning msg="failed to list first level dirs for path: backupstore/volumes reason: Failed to execute: ls [-1 /var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes], output ls: cannot access \'/var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes\': No such file or directory\\n, error exit status 2" pkg=backupstore\ntime="2021-03-08T04:05:20Z" level=error msg="Failed to execute: ls [-1 /var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes], output ls: cannot access \'/var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes\': No such file or directory\\n, error exit status 2"\n, error exit status 1\n{\'code\': 500, \'detail\': \'\', \'message\': \'error listing backups: error listing backup volumes: Failed to execute: /var/lib/longhorn/engine-binaries/c3y1huang-research-8-lh-engine-bcf73c2-dirty/longhorn [backup ls --volume-only nfs://longhorn-test-nfs-svc.default:/opt/backupstore], output Failed to execute: ls [-1 /var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes], output ls: cannot access \\\'/var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes\\\': No such file or directory\\n, error exit status 2\\n, stderr, time="2021-03-08T04:05:20Z" level=warning msg="failed to list first level dirs for path: backupstore/volumes reason: Failed to execute: ls [-1 /var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes], output ls: cannot access \\\'/var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes\\\': No such file or directory\\\\n, error exit status 2" pkg=backupstore\\ntime="2021-03-08T04:05:20Z" level=error msg="Failed to execute: ls [-1 /var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes], output ls: cannot access \\\'/var/lib/longhorn-backupstore-mounts/longhorn-test-nfs-svc_default/opt/backupstore/backupstore/volumes\\\': No such file or directory\\\\n, error exit status 2"\\n, error exit status 1\', \'status\': 500}')
```

Reply:
OK, got it. If the deletion happens in the mid of the 2-level volume directory listing opertions, this race condition will fail the request.

Reply:
Also as part of the backupstore list operation, an absent volumes folder should return an empty result set instead of returning an error.

Reply:
@joshimoo , thanks for clarifying the error returning is unexpected!